### PR TITLE
Clear Ruby warnings

### PR DESCRIPTION
**Pull request by casperisfine**
DNSRuby generates tons of unused variable and other ruby warnings.

That's a problem for applications and libraries depending on it and trying to catch issues on CI/test with these warnings.

This PR clear most warnings, there's still a bunch in the test suite, but I believe it's a good start.

### CODE REVIEW

File: test/spec_helper.rb [hunk 1]
```diff
@@ -1,3 +1,9 @@
+$VERBOSE = true
+
+if Warning.respond_to?(:[]=)
+  Warning[:deprecated] = true
```
Comment (keithrbennett) [inline test/spec_helper.rb:4]: I'm curious, does setting this to true _disable_ all deprecation warnings? If so, would we see these warnings when running `gem build` (I forget)? We should probably see them _somewhere_.

Comment (casperisfine) [reply test/spec_helper.rb:4]: > does setting this to true _disable_ all deprecation warnings?

No, it turns them on actually. Starting from ruby `2.7.2` deprecation warnings are off by default. So it's good practice to turn them on in your test suite. Ultimately the intent is to have test frameworks such as minitest to do it by default, but until then...

**Comment by casperisfine** [comment]
Oh by the way, you don't have a CI anymore as Travis-CI.org is now shut down. Most people migrated to GitHub actions.

**Comment by alexdalitz** [comment]
Nice one - thanks!

**Pull request merged**